src/System.Private.CoreLib/src/Internal/Reflection/Core/NonPortable/IRuntimeImplementedAssembly.cs

```diff
@@ -0,0 +1,11 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Internal.Reflection.Core.NonPortable
+{
+    // Marker interface used to detect whether an Assembly object is one implemented by the core framework.
+    public interface IRuntimeImplementedAssembly
```

Comment:
I am wondering whether we should have just a single `IRuntimeImplemented` type that would be used to mark all runtime-implemented items.

Reply:
Agreed, I was wondering the same. (AFK for another 4 hours, can do it then)